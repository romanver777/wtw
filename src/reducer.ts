import movies from "./mocks/films.json";
import { FilmsType } from "./types/types";
import { GENRES } from "./helpers/const";

interface InitStateType {
  films: FilmsType;
  genre: string;
}

interface ActionType {
  SET_GENRE: string;
  GET_FILMS_BY_GENRE: string;
}

const InitialState: InitStateType = {
  films: [...movies],
  genre: GENRES[0],
};

const ActionType = {
  SET_GENRE: `SET_GENRE`,
  GET_FILMS_BY_GENRE: `GET_FILMS_BY_GENRE`,
};

const ActionCreator = {
  setGenre: (genre: string) => ({
    type: ActionType.SET_GENRE,
    payload: genre,
  }),
  getFIlmsByGenre: (genre, films) => ({
    type: ActionCreator.GET_FILMS_BY_GENRE,
    payload: [],
  }),
};

const reducer = (state = InitialState, action): InitStateType => {
  switch (action.type) {
    case ActionType.SET_GENRE:
      return {
        ...state,
        genre: action.payload,
      };
    default:
      return state;
  }
};

const Operation = {};

export { reducer, ActionCreator, Operation };
