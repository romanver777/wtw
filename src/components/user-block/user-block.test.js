/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import TestRenderer from "react-test-renderer";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { reducer } from "../../reducer";
import { Router } from "react-router-dom";
import history from "../../history";

import UserBlock from "./user-block";

describe("UserBlock component", () => {
  describe("default props", () => {
    const initState = {
      isAuth: false,
    };
    const store = createStore(reducer, initState);
    const Trenderer = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <UserBlock />
        </Router>
      </Provider>
    );
    it("renders correctly", () => {
      expect(Trenderer.toJSON()).toMatchSnapshot();
    });
  });
  describe("isAuth = true", () => {
    const initState = {
      isAuth: true,
    };
    const store = createStore(reducer, initState);
    const Trenderer = TestRenderer.create(
      <Provider store={store}>
        <Router history={history}>
          <UserBlock />
        </Router>
      </Provider>
    );
    const root = Trenderer.root;
    it("renders correctly", () => {
      expect(
        root.findByProps({ className: "user-block__avatar" }).props.children
          .type
      ).toEqual("img");
    });
  });
});
