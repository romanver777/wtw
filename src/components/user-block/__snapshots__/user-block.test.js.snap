// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock component default props renders correctly 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="#/login"
    onClick={[Function]}
  >
    Вход
  </a>
</div>
`;
